fails to build with ghc-8.8

Not completely sure if this is a more general issue but FYI:

In Stackage Nightly (ghc-8.8.1)

```
    Preprocessing library for fakedata-0.3.0..
    Building library for fakedata-0.3.0..
    [  6 of 294] Compiling Config
    
    /var/stackage/work/unpack-dir/unpacked/fakedata-0.3.0-29bee891d16361edbdf0eab1ca8835dc307511faa242308d7b1cdba103cf4eeb/src/Config.hs:387:7: error:
        • Could not deduce (MonadFail m) arising from a use of ‘parseMonad’
          from the context: (MonadThrow m, MonadIO m)
            bound by the type signature for:
                       fetchData :: forall (m :: * -> *) a.
                                    (MonadThrow m, MonadIO m) =>
                                    FakerSettings
                                    -> SourceData -> (FakerSettings -> Value -> Parser a) -> m a
            at src/Config.hs:(369,1)-(374,8)
          Possible fix:
            add (MonadFail m) to the context of
              the type signature for:
                fetchData :: forall (m :: * -> *) a.
                             (MonadThrow m, MonadIO m) =>
                             FakerSettings
                             -> SourceData -> (FakerSettings -> Value -> Parser a) -> m a
        • In a stmt of a 'do' block: parseMonad (parser settings) yaml
          In the expression:
            do afile <- getSourceFile fname
               bs <- liftIO $ BS.readFile afile
               yaml <- decodeThrow bs
               let nhash = HM.insert ckey yaml cache
               ....
          In a case alternative:
              Nothing
                -> do afile <- getSourceFile fname
                      bs <- liftIO $ BS.readFile afile
                      yaml <- decodeThrow bs
                      ....
        |
    387 |       parseMonad (parser settings) yaml
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    
    /var/stackage/work/unpack-dir/unpacked/fakedata-0.3.0-29bee891d16361edbdf0eab1ca8835dc307511faa242308d7b1cdba103cf4eeb/src/Config.hs:408:16: error:
        • Could not deduce (MonadFail m) arising from a use of ‘parseMonad’
          from the context: (MonadThrow m, MonadIO m)
            bound by the type signature for:
                       fetchDataSingle :: forall (m :: * -> *).
                                          (MonadThrow m, MonadIO m) =>
                                          FakerSettings
                                          -> SourceData
                                          -> (FakerSettings -> Value -> Parser Text)
                                          -> m (Vector Text)
            at src/Config.hs:(390,1)-(395,20)
          Possible fix:
            add (MonadFail m) to the context of
              the type signature for:
                fetchDataSingle :: forall (m :: * -> *).
                                   (MonadThrow m, MonadIO m) =>
                                   FakerSettings
                                   -> SourceData
                                   -> (FakerSettings -> Value -> Parser Text)
                                   -> m (Vector Text)
        • In the second argument of ‘(<$>)’, namely
            ‘parseMonad (parser settings) yaml’
          In a stmt of a 'do' block:
            pure <$> parseMonad (parser settings) yaml
          In the expression:
            do afile <- getSourceFile fname
               bs <- liftIO $ BS.readFile afile
               yaml <- decodeThrow bs
               let nhash = HM.insert ckey yaml cache
               ....
        |
    408 |       pure <$> parseMonad (parser settings) yaml
        |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -1,5 +1,9 @@
 # Changelog for fakedata
 
+## 0.3.1
+
+* Make it compatbile with ghc-8.8.1
+
 ## 0.3.0
 
 * Update fake data source

--- a/fakedata.cabal
+++ b/fakedata.cabal
@@ -4,10 +4,10 @@ cabal-version: 1.12
 --
 -- see: https://github.com/sol/hpack
 --
--- hash: be5c05f79b0f39b9b500b1e935b69e331f6bb4f7120972b17688182b7d098a8a
+-- hash: b6a65970d5f1dbdf544a14cc669dc23f816ed4bf82309326c0e5f76a916409f1
 
 name:           fakedata
-version:        0.3.0
+version:        0.3.1
 synopsis:       Library for producing fake data
 description:    Please see the README on GitHub at <https://github.com/psibi/fakedata#readme>
 category:       Random, Fake, FakeData

--- a/package.yaml
+++ b/package.yaml
@@ -1,5 +1,5 @@
 name:                fakedata
-version:             0.3.0
+version:             0.3.1
 github:              "psibi/fakedata"
 license:             BSD3
 author:              "Sibi Prabakaran"

--- a/src/Config.hs
+++ b/src/Config.hs
@@ -384,8 +384,9 @@ fetchData settings sdata parser = do
       yaml <- decodeThrow bs
       let nhash = HM.insert ckey yaml cache
       liftIO $ setCacheFile nhash settings
-      parseMonad (parser settings) yaml
-    Just yaml -> parseMonad (parser settings) yaml
+      either (throwM . ParseError) pure (parseEither (parser settings) yaml)
+    Just yaml ->
+      either (throwM . ParseError) pure (parseEither (parser settings) yaml)
 
 fetchDataSingle ::
      (MonadThrow m, MonadIO m)
@@ -405,5 +406,8 @@ fetchDataSingle settings sdata parser = do
       yaml <- decodeThrow bs
       let nhash = HM.insert ckey yaml cache
       liftIO $ setCacheFile nhash settings
-      pure <$> parseMonad (parser settings) yaml
-    Just yaml -> pure <$> parseMonad (parser settings) yaml
+      pure <$>
+        either (throwM . ParseError) pure (parseEither (parser settings) yaml)
+    Just yaml ->
+      pure <$>
+      either (throwM . ParseError) pure (parseEither (parser settings) yaml)

--- a/src/Faker.hs
+++ b/src/Faker.hs
@@ -39,23 +39,24 @@ import Data.IORef
 import Data.Text (Text)
 import Data.Typeable
 import Data.Vector (Vector)
-import Data.Yaml (Value)
 import Data.Word (Word64)
+import Data.Yaml (Value)
 import Faker.Internal.Types (CacheFieldKey, CacheFileKey)
-import System.Random (StdGen, newStdGen, split, mkStdGen)
+import System.Random (StdGen, mkStdGen, newStdGen, split)
 
-data FakerSettings =
-  FakerSettings
-    { fslocale :: !Text -- ^ Locale settings for your fake data source.
-    , fsrandomGen :: !StdGen -- ^ Seed to initialize random generator state
-    , fsDeterministic :: !Bool -- ^ Controls whether you want
+data FakerSettings = FakerSettings
+  { fslocale :: !Text -- ^ Locale settings for your fake data source.
+  , fsrandomGen :: !StdGen -- ^ Seed to initialize random generator state
+  , fsDeterministic :: !Bool -- ^ Controls whether you want
                             -- deterministic out. This overrides
                             -- 'fsrandomGen'.
-    , fsCacheField :: (IORef (HM.HashMap CacheFieldKey (Vector Text)))
-    , fsCacheFile :: (IORef (HM.HashMap CacheFileKey Value))
-    }
+  , fsCacheField :: (IORef (HM.HashMap CacheFieldKey (Vector Text)))
+  , fsCacheFile :: (IORef (HM.HashMap CacheFileKey Value))
+  }
 
-newtype FakerGen = FakerGen { unFakerGen :: (Int, StdGen)} deriving (Show)
+newtype FakerGen = FakerGen
+  { unFakerGen :: (Int, StdGen)
+  } deriving (Show)
 
 instance Show FakerSettings where
   show (FakerSettings {..}) =
@@ -65,12 +66,16 @@ data FakerException
   = InvalidLocale String -- ^ This is thrown when it is not able to
                          -- find the fake data source for your
                          -- localization.
-  | InvalidField String Text -- ^ The 'String' represents the field it is
+  | InvalidField String
+                 Text -- ^ The 'String' represents the field it is
                              -- trying to resolve and the 'Text' field
                              -- is something you passed on.
   | NoDataFound FakerSettings -- ^ This is thrown when you have no
                               -- data. This may likely happen for
                               -- locales other than `en`.
+  | ParseError String -- ^ This is thrown when the parsing step
+                      -- fails. The 'String' represents the error
+                      -- message.
   deriving (Typeable, Show)
 
 instance Exception FakerException
@@ -167,10 +172,9 @@ replaceCacheFile cache fs = do
 
 -- | Fake data type. This is the type you will be using to produce
 -- fake values.
-newtype Fake a =
-  Fake
-    { unFake :: FakerSettings -> IO a
-    }
+newtype Fake a = Fake
+  { unFake :: FakerSettings -> IO a
+  }
 
 instance Functor Fake where
   fmap :: (a -> b) -> Fake a -> Fake b

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,4 @@
-resolver: lts-13.9
+resolver: nightly-2019-10-06
 
 packages:
 - .

